clang/test/SemaSYCL/attr-add-ir-attributes-merge.cpp

```diff
@@ -0,0 +1,53 @@
+// RUN: %clang_cc1 -fsycl-is-device -std=gnu++11 -fsyntax-only -verify %s
+
+// Tests that add_ir_attributes_* allows merging when the filter and IR
+// attributes to generate are the same.
+
+void FunctionRedecl1();
+[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl1();
+[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl1();
+[[__sycl_detail__::add_ir_attributes_function("Attr2", "Attr1", true, 1)]] void FunctionRedecl1(); // expected-note {{conflicting attribute is here}}
+[[__sycl_detail__::add_ir_attributes_function("Attr3", false)]] void FunctionRedecl1(){};          // expected-error {{attribute 'add_ir_attributes_function' is already applied with different arguments}}
+
+[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl2();
+void FunctionRedecl2();
+
+void FunctionRedecl3();
+[[__sycl_detail__::add_ir_attributes_function({"Attr1", "Attr3"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3();
+[[__sycl_detail__::add_ir_attributes_function({"Attr1", "Attr3"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3();
+[[__sycl_detail__::add_ir_attributes_function({"Attr1", "Attr3"}, "Attr2", "Attr1", true, 1)]] void FunctionRedecl3();
+[[__sycl_detail__::add_ir_attributes_function({"Attr3", "Attr1"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3(); // expected-note {{conflicting attribute is here}}
+[[__sycl_detail__::add_ir_attributes_function({"Attr1"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3();          // expected-error {{attribute 'add_ir_attributes_function' is already applied with different arguments}} expected-note {{conflicting attribute is here}}
+[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl3(){};                   // expected-error {{attribute 'add_ir_attributes_function' is already applied with different arguments}}
```

Comment:
Wait. Why is this an error? Is it because filter list isn't present?

Reply:
Yes, exactly. Similar to https://github.com/intel/llvm/pull/5535#discussion_r825027061, the filters must currently have the exact same values, so a missing filter is not considered the same as `{"Attr1", "Attr3"}`.